### PR TITLE
Task5 done

### PR DESCRIPTION
**Pull request by stefera**

### CODE REVIEW

File: CONTROLLER errors.js [hunk 1]
```diff
@@ -1,3 +1,19 @@
+module.exports = handle400 = (err, request, response, next) => {
+  if (err.code) {
+    response.status(400).send({ message: "Invalid request, try again" });
+  } else {
+    next(err);
+  }
+};
+
+module.exports = handle404 = (err, request, response, next) => {
+  // console.log("error arrived", err);
```
Comment (Jodiech) [inline CONTROLLER errors.js:10]: Please remove console logs 👍

File: CONTROLLER.js [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+};
+
+const getArticleById = (request, response, next) => {
+  console.log(request.params);
+  const { articleId } = request.params;
+  // console.log("articleIDdestructured", articleId);
+  fetchArticleById(articleId)
+    .then((selectedArticle) => {
+      // console.log("arrived in response", { article: selectedArticle });
+      response.status(200).send({ article: selectedArticle });
```
Comment (Jodiech) [inline CONTROLLER.js:42]: Great choice of key to send articles back 👍

File: __tests__/APP.test.js [hunk 3]
```diff
@@ -61,3 +61,59 @@ describe("task3- GET api/topics", () => {
...
+
+describe.only("task5- GET api/articles/:articleiD", () => {
+  test("returns a single object (with correct properties article) when id exists", () => {
+    return request(app)
+      .get("/api/articles/3")
+      .expect(200)
+      .then(({ body }) => {
+        // console.log("in test", body.article);
+        const article = body.article;
+        expect(article).toMatchObject({
```
Comment (lewispricey) [inline __tests__/APP.test.js:152]: This is a great test however you're missing a check for the votes property (this is key as we use likes further down the line in the backend and use the like/votes functionality on the front end)

File: MODELS.js [hunk 4]
```diff
@@ -16,4 +16,63 @@ const fetchTopics = () => {
...
+  return db
+    .query(
+      `
+       SELECT * FROM articles
+       WHERE article_id = ${article_id}
+       `
+    )
+    .then(({ rows }) => {
+      // console.log("rows of article", rows[0]);
+      if (!rows[0]) {
```
Comment (Jodiech) [inline MODELS.js:63]: Nice error handling here 😄 please remove your console logs

File: MODELS.js [hunk 5]
```diff
@@ -16,4 +16,63 @@ const fetchTopics = () => {
...
+       GROUP BY articles.article_id       
+       ORDER BY created_at DESC
+       `
+    )
+    .then(({ rows }) => {
+      return rows;
+    });
+};
+
+const fetchCommentCount = (arrayOfArticles) => {
```
Comment (lewispricey) [inline MODELS.js:40]: Is this function still required?

File: __tests__/APP.test.js (hunk 3 above)
Comment (Jodiech) [inline __tests__/APP.test.js:152]: Great to see you have used a flexible matcher here with toMatchObject. This allows for later extension of the endpoint. 👍

File: MODELS.js [hunk 6]
```diff
@@ -16,4 +16,63 @@ const fetchTopics = () => {
...
+    )
+    .then(({ rows }) => {
+      // console.log("rows of article", rows[0]);
+      if (!rows[0]) {
+        return Promise.reject({
+          status: 404,
+          msg: "Item not found, try again",
+        });
+      }
+      return rows[0];
```
Comment (lewispricey) [inline MODELS.js:69]: Really like that you're just returning the first item from the response array here so whatever goes back to the user is an object rarther than an array 💯

File: __tests__/APP.test.js [hunk 7]
```diff
@@ -61,3 +61,59 @@ describe("task3- GET api/topics", () => {
       });
   });
 });
+
+describe.only("task5- GET api/articles/:articleiD", () => {
```
Comment (Jodiech) [inline __tests__/APP.test.js]: Please remove the .only for this test.

**Review by Jodiech — APPROVED** [review]
Nice work Sam! I have left a few comments below, mainly for tidying up your code and removing console logs and .only. Please merge once you have made these changes 😄

**Review by lewispricey — APPROVED** [review]
Great work on this ticket Samuel!

Please add in the assertion to ensure you're recieving the votes property back on the object.

Theres lots of console logs and commented out code in this ticket, please make sure to remove all of this before merging.

No need to raise another PR just please add in the votes before you merge as without this the front end will be very difficult! 🚀

**Pull request merged**